### PR TITLE
feature/CVSB-11570 - [FE] As a VTM user,  I want to be able to navigate to the Test Details screen

### PR DESCRIPTION
**Pull request by irina005**
**Needs to be updated for systemNumber changes** 

https://jira.dvsacloud.uk/browse/CVSB-11570

### CODE REVIEW

File: src/app/test-record/test-record.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@Component({
+  selector: 'vtm-test-record',
+  templateUrl: './test-record.component.html',
+  styleUrls: ['./test-record.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TestRecordComponent implements OnInit {
+
+  testResultObservable$: Observable<any>;
+  testType: any;
```
Comment (m-ujaffer) [inline src/app/test-record/test-record.component.ts]: specify the type to be `TestType`

File: src/app/test-record/test-record.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component({
+  selector: 'vtm-test-record',
+  templateUrl: './test-record.component.html',
+  styleUrls: ['./test-record.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TestRecordComponent implements OnInit {
+
+  testResultObservable$: Observable<any>;
```
Comment (m-ujaffer) [inline src/app/test-record/test-record.component.ts]: specify the type to be `TestResultModel`

File: src/app/technical-record/test-history/test-history.component.html [hunk 3]
```diff
@@ -6,27 +6,37 @@
...
-    </td>
-    <td id="{{ 'test-testResult-' + i }}" class="govuk-table__cell govuk-!-font-weight-bold"
-        [ngStyle]="testRes.testTypes[0].testResult=='pass' ? { color:'#00703C' } : testRes.testTypes[0].testResult=='prs' ? {color: '#1D70B8'} : (testRes.testTypes[0].testResult=='fail' || testRes.testTypes[0].testResult=='abandoned') ? {color: '#D4351C'} : {color: ''}">
-      {{ (testRes.testTypes[0].testResult | DefaultNullOrEmpty) | uppercase }}
-    </td>
-  </tr>
+  <ng-container *ngFor="let testRes of testResultJson">
+    <tr class="govuk-table__row" *ngFor="let testType of testRes.testTypes; let i = index">
+      <td id="{{ 'test-testCode-' + i }}" class="govuk-table__cell">
+        {{ (testType.testCode | DefaultNullOrEmpty).charAt(0).toUpperCase() + (testType.testCode | DefaultNullOrEmpty).substr(1).toLowerCase() }}
```
[the remark this answers is not in the document]
Comment (irina005) [reply src/app/technical-record/test-history/test-history.component.html]: Yes, it is, but at the time I worked on this, the branch with the change wasn't merged into develop.

File: src/app/app.module.ts [hunk 4]
```diff
@@ -61,6 +62,7 @@ export const COMPONENTS = [
   FooterComponent,
   TechnicalRecordCreateComponent,
   SpinnerLoaderComponent,
+  TestRecordComponent
```
[the remark this answers is not in the document]
Comment (irina005) [reply src/app/app.module.ts]: I created a test-record module into 10283 branch, which is the second part of functionality

[the remark this answers is not in the document]
Comment (irina005) [reply src/app/app.module.ts]: image

File: src/app/test-record/test-record.component.ts (hunk 1 above)
Comment (irina005) [reply src/app/test-record/test-record.component.ts]: Fixed into 10283, but I can fix in here also

**Pull request merged**